jsonschema-core.xml

```diff
@@ -1981,6 +1984,8 @@
                         <t>
                             An instance validates successfully against this keyword if it validates
                             successfully against at least one schema defined by this keyword's value.
+                            Note that when annotations are being collected, they MUST be collected
+                            from all subschemas if any subschema validates successfully.
```

Comment:
But wouldn't annotations from failed subschemas be thrown away?

```suggestion
                            from any subschemas that validate successfully.
```

Reply:
@gregsdennis yes?  That doesn't mean they aren't collected.  Yes, once you know a subschema has failed you can stop collecting, but you collect until you figure that out, *THEN* drop the annotations.

I'm really frustrated by my apparent inability to get this across.  It's discussed all over the spec- is there some magic phrase you're looking for?

Reply:
If I conform to "from **all** subschemas if any subschema validates successfully," then when any subschema passes validation, then I get annotations from **all** subschemas, whether or not they pass.  My wording requires that annotations only from passing subschemas are kept.

> Yes, once you know a subschema has failed you can stop collecting...

I think you mean "passed" instead of "failed."  This is `anyOf` not `allOf`.  I read what you mean, though.

My focus for this is the result, not the process.  It seems that it should be understood that an implementation has to collect until validation is certain, but I don't think we have to explain that process here.  I think it's sufficient to say that we want the annotations from all passing subschemas.  We definitely *don't* want annotations from failing subschemas, which is what your wording here tells me to do.

Reply:
@gregsdennis I see your point here.

What I'm trying to get across is that you must _perform the process of collecting annotations_ on all of the subschemas if any of them validate successfully.  Since part of that process is dropping annotations from any subschema that fails to validate, performing the process does not mean that you will get annotations from every subschema.

So we clearly need some better wording around that.  I see the point of your suggestion and it is probably better, but let me see if I can come up with something that is more clear about both aspects.

Reply:
@gregsdennis OK I pushed a new commit with a change to this line and another one- you should probably look at that commit's diff separately to see where they both are.

I tried to make a distinction between _examining_ a subschema (which may or may not result in actual annotations depending on its assertion outcome) and actually keeping the collected annotations.

Reply:
That's a lot better.